### DISCUSSION TITLE
An error "Illegal instruction" on `poetry add` command

### DISCUSSION THREAD

**Discussion by GusarevKirill** [python-poetry/poetry#9897] [Q&A] [answered]
### Problem description
Hi! I am a little developer and I moved my python project from Windows to Ubuntu and tried to run a command `poetry install`, but an error "Illegal instruction (core dumped)" was given. I receive the same error if I try to run `poetry add` with any package.
[Error screenshot](https://i.imgur.com/SXwuk7Z.png)

### Machine specification
Python: 3.12.3
Poetry: 1.8.5 (installed via curl)
OS: Ubuntu 24.04.1 LTS
CPU: Intel Atom D510 1.66 GHz (2 cores 4 threads, x86-64)
RAM: 3 GB

Help me to solve this problem please :[

**Comment by Secrus** [comment]
There isn't much I could say about the issue without more error logs, but my first instinct is that either your machine has some weird quirks or one of your dependencies has compiled extensions and they do something weird in the compilation step. 
You could also try `poetry config installer.max-workers 1` to make sure it's not a threading issue.

**Reply by GusarevKirill** [reply]
Now I tried to `poetry add click` after `poetry config installer.max-workers 1` but received the same error :(
[Screenshot of same error with `-vvv` option](https://i.imgur.com/Op2xI3B.png)
I say second time that I receive this error with ANY dependencies (any PYPI packets).
And installing packets via pip works nice!
Problem is in poetry. I think that Poetry uses some new CPU instructions (which my CPU can't understand)... What do you think about it?

**Reply by Secrus** [reply, marked as answer]
Try setting `poetry config keyring.enabled false`, maybe that's something that is failing on your machine.

**Reply by GusarevKirill** [reply]
> Try setting `poetry config keyring.enabled false`, maybe that's something that is failing on your machine.

Yes, this helped! Thank you very much! ^_^
But what is a "keyring" and what does it do for poetry?
What are the disadvantages of disabled keyring?

**Reply by Secrus** [reply]
The keyring is used to interact with your system credential store.

**Reply by GusarevKirill** [reply]
> The keyring is used to interact with your system credential store.

Ah, well I don't need that)
Thank you!
